pkg/roachprod/vm/gce/dns.go

```diff
@@ -439,3 +443,88 @@ func (p *dnsProvider) syncPublicDNS(l *logger.Logger, vms vm.List) (err error) {
 		cmd, output, zoneBuilder.String(),
 	)
 }
+
+// ConfigureDnsHost configures the DNS host for the cluster based on the action
+func (p *dnsProvider) ConfigureDnsHost(
+	ctx context.Context, clusterName, action string, hosts []string,
+) error {
+	for hostIndex, host := range hosts {
+		matches := hostIPRegExp.FindStringSubmatch(host)
+		if len(matches) < 2 {
+			return errors.Newf("host %s could not be parsed for IP.", host)
+		}
+		ip := matches[1]
+		name := fmt.Sprintf("%s-%04d.%s.", clusterName, hostIndex+1, p.publicDomain)
```

Comment:
I wonder if we should maybe try to re-use `PublicDNS` on the VM struct to avoid code duplication. It might be better to pass a list of IPs + PublicDNS info to this method.